### PR TITLE
htlcswitch: remove lnrpc dependency

### PR DESCRIPTION
**Pull request by joostjager**
Removes an unwanted dependency that prevents moving parts of rpcserver to a sub server.

### CODE REVIEW

File: htlcswitch/switch.go [hunk 1]
```diff
@@ -101,6 +100,18 @@ const (
 	CloseBreach
 )
 
+// PendingUpdate describes the pending state of a closing channel.
+type PendingUpdate struct {
+	Txid        []byte
```
Comment (Roasbeef) [inline htlcswitch/switch.go]: Missing `godoc` field comments here and below.

File: lnrpc/rpc.proto [hunk 2]
```diff
@@ -1153,13 +1153,6 @@ message GetInfoResponse {
     uint32 num_inactive_channels = 15 [json_name = "num_inactive_channels"];
 }
 
-message ConfirmationUpdate {
```
Comment (Roasbeef) [inline lnrpc/rpc.proto]: The message currently isn't used, but recently we've gained the ability to actually populate it. Would prefer to leave it in tact (but possibly remove the entry in the oneof so the docs are accurate).

File: htlcswitch/switch.go [hunk 3]
```diff
@@ -101,6 +100,18 @@ const (
...
+	OutputIndex uint32
+}
+
+// ChannelCloseUpdate contains the outcome of the close channel operation.
```
Comment (Roasbeef) [inline htlcswitch/switch.go]: FWIW, we've added two new structs to this package in this PR, yet they aren't referenced at all in the package. Instead, the structs are only created in the peer/rpcserver. This indicates that perhaps these structs shouldn't live in this package at all, or even a more aggressive refactoring is warranted.

File: htlcswitch/switch.go (hunk 1 above)
Comment (joostjager) [reply htlcswitch/switch.go]: Moved structs and made internal.

File: lnrpc/rpc.proto (hunk 2 above)
Comment (joostjager) [reply lnrpc/rpc.proto]: Reverted delete.

File: htlcswitch/switch.go (hunk 3 above)
Comment (joostjager) [reply htlcswitch/switch.go]: I moved them into `peer.go` where they are primarily used. More aggressive refactoring than this (for example creating a different pathway than that `interface{}` channel), I'd like to leave for another time. This PR is already a sub sub task.

Comment (Roasbeef) [reply htlcswitch/switch.go]: SGTM

**Pull request merged**